source/lac/sparsity_tools.cc

```diff
@@ -326,6 +331,66 @@ namespace SparsityTools
   }
 
 
+  void color_sparsity_pattern (const SparsityPattern     &sparsity_pattern,
+                               std::vector<int> &color_indices)
+  {
+    // Make sure that ZOLTAN is actually
+    // installed and detected
+#ifndef DEAL_II_TRILINOS_WITH_ZOLTAN
+    (void)sparsity_pattern;
+    AssertThrow (false, ExcZOLTANNotInstalled());
+#else
+    //Zoltan coloring algorithm is run in series by each processor. All objects colored are assumed to belong to Processor 0.
+    std::unique_ptr<Zoltan> zz = std_cxx14::make_unique<Zoltan> (MPI_COMM_SELF);
+
+    //Coloring parameters
+    zz->Set_Param ("COLORING_PROBLEM", "DISTANCE-1");  //Choosing standard vertex problem.
+    zz->Set_Param ("NUM_GID_ENTRIES", "1");  // global ID is represented by 1 integer
+    zz->Set_Param ("NUM_LID_ENTRIES", "1");  // local ID is represented by 1 integer. Not used, but required by "Zoltan::Color" function
+    zz->Set_Param ("OBJ_WEIGHT_DIM", "0");   // object weights not used
+    zz->Set_Param ("RECOLORING_NUM_OF_ITERATIONS", "0");
+    zz->Set_Param ("DEBUG_LEVEL", "0" );     //level of debug info
+
+    //Zoltan::Color function needs a non-const object equal to sparsity_pattern, which is a const object
+    SparsityPattern graph;
+    graph.copy_from (sparsity_pattern);
+
+    //Set query functions required by coloring function
+    zz->Set_Num_Obj_Fn (get_number_of_objects, &graph);
+    zz->Set_Obj_List_Fn (get_object_list, &graph);
+    zz->Set_Num_Edges_Multi_Fn (get_num_edges_list, &graph);
+    zz->Set_Edge_List_Multi_Fn (get_edge_list, &graph);
+
+    //Variables needed by coloring function
+    int num_gid_entries=1;
+    const int num_objects = graph.n_rows();
+    ZOLTAN_ID_PTR global_ids = new ZOLTAN_ID_TYPE [num_objects];   //Pointer type fixed by Color function. Size needs to be preallocated.
```

Comment:
can't you use `std::vector< ZOLTAN_ID_PTR> global_ids(num_objects);` and avoid manually deleting? Same for `color_exp`.